ehrapy/tools/nlp/_medcat.py

```diff
@@ -1,53 +1,74 @@
 from __future__ import annotations
 
-import warnings
-from dataclasses import dataclass
-from typing import Literal
-
-import numpy as np
 import pandas as pd
-import seaborn as sns
 from anndata import AnnData
-from matplotlib import pyplot as plt
-from medcat.cat import CAT
-from medcat.cdb import CDB
-from medcat.cdb_maker import CDBMaker
-from medcat.config import Config
-from medcat.vocab import Vocab
-from rich import print
-from rich.progress import Progress, SpinnerColumn
-from spacy import displacy
-from spacy.tokens.doc import Doc
-
-from ehrapy import settings
-from ehrapy.core.tool_available import check_module_importable
-
-spacy_models_modules: list[str] = list(map(lambda model: model.replace("-", "_"), ["en-core-web-md"]))
-for model in spacy_models_modules:
-    if not check_module_importable(model):
-        print(
-            f"[bold yellow]Model {model} is not installed. Refer to the ehrapy installation instructions if required."
-        )
-
 
-@dataclass
-class AnnotationResult:
-    all_medcat_annotation_results: list | None
-    entities_pretty: dict[str, list[str]]
-    cui_locations: dict | None
-    type_ids_location: dict | None
+try:
+    from medcat.cat import CAT
+    from medcat.cdb import CDB
+    from medcat.cdb_maker import CDBMaker
+    from medcat.config import Config
+    from medcat.vocab import Vocab
+except ModuleNotFoundError:
```

Comment:
The issue why I introduced this in the first place was that some tests were failing for unknown reasons because medcat has not been found (I double checked in the env and verified import was working in the REPL).
Maybe this is due to medcat being an extra dependency (which we should maybe reconsider).

Reply:
We should be able to make this work. medcat should stay an extra dependency imo because it has `transformers` as a dependency which then pulls tensorflow and jax. This can get messy very quickly.